packages/core/src/errors/InruptError.ts

```diff
@@ -21,60 +21,107 @@
 
 /**
  * @hidden
- * Extends the regular JavaScript error object with extra meta-data (e.g.
+ * The generic Inrupt error class, that simply extends the regular JavaScript
+ * [[Error]] object but provides extra capabilities and meta-data (e.g.
  * potentially data from a HTTP response, or with data from an RDF vocabulary
  * (which may be imported locally, or looked up dynamically at runtime)).
+ *
+ * Error Identifiers
+ *   Our error class supports the notion of globally unique, dereferenceable
+ *   error identifiers in the form of IRIs.
+ *
+ * Parameterized error messages
+ *   This class supports error messages with positional placeholders that can be
+ *   replaced by developer-provided parameter values at runtime.
+ *
+ * Wrapped Errors (Coming soon!)
+ *   Similar to how Java supports the notion of wrapped exceptions, our error
+ *   class implementation supports the ability to wrap [[Error]]s.
+ *
+ * HTTP errors:
+ *   This class provides a very convenient means to provide a HTTP response,
+ *   the details from which can be appended to error message strings, and the
+ *   response itself retrieved directly by consumers of instances of this class.
  * @packageDocumentation
  */
 
-/**
- * In a subsequent PR, just re-export needed implementations from generated
- * artifacts directly, e.g. like @inrupt/solid-common-vocab, e.g. in the
- * `index.hbs` template of RDF/JS artifacts include this:
- // Re-export some of the basic RDF/JS implementations that we use ourselves -
- // allows users of our generated artifacts to easily import these
- // implementations if they want (e.g. in their tests, or to instantiate
- // hard-coded IRIs for convenience (as they shouldn't really need to do that
- // much, instead using our generated constants in most cases)).
- export { NamedNode } from "rdf-js";
- *  Then we can either import that implementation in this code, or just import
- *  the types when needed here, e.g.:
- import type { NamedNode } from "rdf-js";
- */
-import { NamedNode } from "n3";
-
-import { VocabTerm } from "@inrupt/solid-common-vocab";
+import { VocabTerm, NamedNode } from "@inrupt/solid-common-vocab";
 
 export default class InruptError extends Error {
```

Comment:
The error type looks to have a lot of HTTP specific details in it. It might be worth considering renaming this to HttpError. InruptError seems very general.

Reply:
Yeah, it's still early days for this class (i.e. no nested error support yet, no runtime dynamic error IRI lookup/retrieval/parsing, etc.), so maybe it looks like there's a lot of HTTP-specific 'stuff' in here. But the intent is just to have this generic InruptError provide **_support for_** HTTP errors responses, that's all - i.e. it's just **_one of_** the capabilities or features that the generic error class provides (e.g. along with nesting errors, looking up IRI's, doing positional parameter replacement, etc.). Does that make sense, or would you still feel that a completely separate HttpError class would be better...?

Reply:
My prefered option would be to keep all the HTTP specific stuff out of the Error entirely. This InruptError would be more useful if it just makes it easy to gather as much context information as possible. So wrapping nested errors, an IRI for the type or Error, plus holding other context information all makes sence here. Handling (or knowing anything about) HTTP or doing lookups (to me) sounds like the wrong place for this functionality. That functionality should be handled in the Error Handling layer.

Reply:
I agree that it feels a little too specific that the generic Inrupt error class has specific HTTP properties. What about creating some types that could be combined to the class, a bit like we do in `solid-client`, something like:
```
const type WithHttpError = {
  httpErrorResponse?: Response & { ok: false };
}

function buildHttpError(error: InruptError, httpError: Response & { ok: false }): InruptError & WithHttpError {
  return ...;
}
```
so that you could do 
```
throw buildHttpError(myInruptError, myHttpError)
```
 ?

I think this would go in the direction of a decorator pattern, but a builder approach is also something we could look at. This approach I think would make it relatively easy to wrap errors too, because all the wrapping logic could be in the `InruptError` class, and the context-specific information would go in the decorated types.

Reply:
Yeah, I agree with Jarlath - adding in the HTTP stuff was really just to support the current Error use-cases in this `solid-client-authn-js` library, and it doesn't really belong here. I totally agree with the generic InruptError basically just being a dumping ground for 'as much context information as possible' (which of course could include HTTP information), but I agree we should use wrappers/decorators/builders to dump that information (and since we can use an RDF dataset as the generic 'information dump', we can keep this class completely agnostic to the data it gathers together). We could then define HTTP-specific, or SOAP-specific, or Apache-Camel-specific classes to transform their domain-specific data into generic RDF to be dumped in here.

Reply:
And yeah Nic, we could create types like you suggest too, that would certainly be one option.